### DISCUSSION TITLE
How to integration with react hook form

### DISCUSSION THREAD

**Discussion by hungify2022** [remix-run/react-router#10347] [Q&A]
Thanks for some cool features in v6.4.

I am trying to use `Form` and `action` with `react hook form` but I don't know a way to submit and to get data `request.formData()` in action.

I tried with `useSubmit` but `formData` is `undefined`

```
export async function action({ request }) {
  const formData = await request.formData();
  const contact = Object.fromEntries(formData);
  console.log('contact', contact); // undefined
}

const {
    register,
    handleSubmit,
  } = useForm();

const onSubmit = async (data, evt) => {}

 <Form
      method="post"
      id="contact-form"
      onSubmit={handleSubmit(onSubmit)}
>
... 
</Form>

```

**Comment by BigSillyTiger** [comment]
maybe this could give you some idea [https://github.com/orgs/react-hook-form/discussions/9910](url)
but I was stuck at: how to use `useSubmit` handling multiple actions with one router
